webhook/psbinding/psbinding.go

```diff
@@ -221,7 +222,10 @@ func (ac *Reconciler) Admit(ctx context.Context, request *admissionv1.AdmissionR
 	if ac.WithContext != nil {
 		var err error
 		ctx, err = ac.WithContext(ctx, fb)
-		if err != nil {
+		if apierrs.IsNotFound(err) {
+			//Allow request for now as the resource might not exist yet
+			return &admissionv1.AdmissionResponse{Allowed: true}
+		} else if err != nil {
```

Comment:
It seems odd to me to key off of `apierrs` since that is generally used for errors returned from API calls, and `WithContext` is an arbitrary callback.

I wonder whether this would be a reasonable use of wrapped errors (paging @n3wscott who seems to love those)?  Essentially what I'm wondering (thinking out loud) is whether we should provide a way for implementors of `WithContext` to wrap the error they return to guide the behavior here in a way that's fitting for the abstraction.

This would give the implementor of `WithContext` the capacity to pick and choose what error types in what contexts result in this being allowed or denied.

Reply:
@n3wscott any comments ?

Reply:
Where is the change that changes WithContext?

Reply:
it was merged in https://github.com/knative/pkg/pull/1688

Reply:
I agree with matt, I think the simple fix could be a better name `WithContext` there is a lot of code that is running there and it is not at all clear that it will be happening because of a `WithContext` call